formats/json-jackson/src/main/java/io/cloudevents/jackson/CloudEventDeserializer.java

```diff
@@ -134,7 +134,17 @@ public JsonMessage(JsonParser p, ObjectNode node) {
                             writer.withContextAttribute(extensionName, extensionValue.booleanValue());
                             break;
                         case NUMBER:
-                            writer.withContextAttribute(extensionName, extensionValue.numberValue());
+
+                            final Number numericValue = extensionValue.numberValue();
+
+                            // Only 'Int' values are supported by the specification
+
+                            if (numericValue instanceof Integer){
+                                writer.withContextAttribute(extensionName, (Integer) numericValue);
+                            } else{
+                                throw CloudEventRWException.newInvalidAttributeValue(extensionName,extensionValue, null);
```

Comment:
Should it be https://github.com/cloudevents/sdk-java/blob/master/api/src/main/java/io/cloudevents/rw/CloudEventRWException.java#L133?

Reply:
doh... probably..

Reply:
Actually, having thought a bit more, maybe the one I chose is more appropriate in this situation as it will actually cause the problematic value to be logged along with the attribute name. The other exception only reports the problematic attribute name.

I'm now leaning to leaving it as-is ... thoughts ?!?

Reply:
For me you should use invalid type as exception kind... Maybe you need another constructor for it that takes and logs the value?

Reply:
You can have an overload like:

```
    public static CloudEventRWException newInvalidAttributeType(String attributeName, Object value) {
        return new CloudEventRWException(
            CloudEventRWExceptionKind.INVALID_ATTRIBUTE_TYPE,
            "Invalid attribute/extension type for \"" + attributeName + "\". Type: " + value.getClass().getCanonicalName() + ". Value: " + value
        );
    }
```

Reply:
Ack .. will do